modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgniteSqlValidator.java

```diff
@@ -62,6 +63,22 @@
     /** Decimal of Integer.MAX_VALUE for fetch/offset bounding. */
     private static final BigDecimal DEC_INT_MAX = BigDecimal.valueOf(Integer.MAX_VALUE);
 
+    /** **/
+    private static final EnumSet<SqlKind> HUMAN_READABLE_ALIASES_FOR;
+
+    static {
+        EnumSet<SqlKind> kinds = EnumSet.noneOf(SqlKind.class);
+
+        kinds.addAll(SqlKind.AGGREGATE);
+        kinds.addAll(SqlKind.BINARY_ARITHMETIC);
+        kinds.addAll(SqlKind.FUNCTION);
+
+        kinds.add(SqlKind.CEIL);
+        kinds.add(SqlKind.FLOOR);
+
+        HUMAN_READABLE_ALIASES_FOR = kinds;
```

Comment:
it might be worth to wrap `kinds` with `Collections.unmodifiableSet()`

Reply:
also it seems that case with simple literals is not covered yet.

`select 1, -1, 'some string' from MyTable`

BTW do we plan to limit length of the aliases?

Reply:
good catch!
fixed first two suggestion

Reply:
added limit length for aliases